pkg/api/db.go

```diff
@@ -177,31 +140,75 @@ func CreateVolumeSnapshotDBEntry(ctx *c.Context, in *model.VolumeSnapshotSpec) (
 	if in.Id == "" {
 		in.Id = uuid.NewV4().String()
 	}
-
 	if in.CreatedAt == "" {
 		in.CreatedAt = time.Now().Format(constants.TimeFormat)
 	}
 
-	var snap = &model.VolumeSnapshotSpec{
-		BaseModel: &model.BaseModel{
-			Id:        in.Id,
-			CreatedAt: in.CreatedAt,
-		},
-		ProfileId:   in.ProfileId,
-		Name:        in.Name,
-		Description: in.Description,
-		VolumeId:    in.VolumeId,
-		Size:        vol.Size,
-		Metadata:    utils.MergeStringMaps(in.Metadata, vol.Metadata),
-		Status:      model.VolumeSnapCreating,
-	}
+	in.Status = model.VolumeSnapCreating
+	return db.C.CreateVolumeSnapshot(ctx, in)
+}
 
-	result, err := db.C.CreateVolumeSnapshot(ctx, snap)
+func CreateReplicationDBEntry(ctx *c.Context, in *model.ReplicationSpec) (*model.ReplicationSpec, error) {
+	pVol, err := db.C.GetVolume(ctx, in.PrimaryVolumeId)
+	if err != nil {
+		log.Error("Get primary volume failed in create volume replication method: ", err)
+		return nil, err
+	}
+	if pVol.Status != model.VolumeAvailable && pVol.Status != model.VolumeInUse {
+		var errMsg = fmt.Errorf("Only the status of primary volume is available or in-use, the replicaiton can be created")
+		log.Error(errMsg)
+		return nil, errMsg
+	}
+	sVol, err := db.C.GetVolume(ctx, in.SecondaryVolumeId)
 	if err != nil {
-		log.Error("Error occurred in dock module when create volume snapshot in db:", err)
+		log.Error("Get secondary volume failed in create volume replication method: ", err)
```

Comment:
Starting letter in error message should be lower case.  Same in all other places.